From 427470eba81992249f0003641df52e6c87090e73 Mon Sep 17 00:00:00 2001
From: Maxim Anisimov <maxim.anisimov.ua@gmail.com>
Date: Mon, 20 Sep 2021 09:50:18 +0300
Subject: [PATCH 2/3] ubusd location

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fc480b6..a885c69 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -79,7 +79,7 @@ INSTALL(TARGETS ubus cli
 	RUNTIME DESTINATION bin
 )
 INSTALL(TARGETS ubusd
-	RUNTIME DESTINATION sbin
+	RUNTIME DESTINATION bin
 )
 
 INSTALL(FILES ubusmsg.h ubus_common.h libubus.h DESTINATION include)
-- 
2.33.0

